### ISSUE TITLE
MSI GL62 6QD laptop: Fan is continuously running

### ISSUE THREAD

**Issue by darknyaz** [Bumblebee-Project/bbswitch#152]
After installing bumblebee with nvidia-driver on my laptop fan is running continuously. Bumblebee itself seems to work: card is on when application is running via optirun and off when application is closed. I also needed to add `acpi_osi=! "acpi_osi=Windows 2009"` options because of screen freezes, `pci=noaer` because of AER flood (I know that it isn't good solution) and `acpi_enforce_resources=lax` trying to solve my fan problem.

`dmesg | grep -C 10 bbswitch:` output:
> [   21.273343] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.338573] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.338813] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.353825] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
> [   21.363621] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.363861] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.428559] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.428812] iwlwifi 0000:02:00.0: L1 Disabled - LTR Enabled
> [   21.442850] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
> [   21.782161] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
> [   22.042583] bbswitch: loading out-of-tree module taints kernel.
> [   22.043223] bbswitch: version 0.8
> [   22.043298] bbswitch: Found integrated VGA device 0000:00:02.0: \_SB_.PCI0.GFX0
> [   22.043426] bbswitch: Found discrete VGA device 0000:01:00.0: \_SB_.PCI0.PEG0.PEGP
> [   22.043571] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20160831/nsarguments-95)
> [   22.044038] bbswitch: detected an Optimus _DSM function
> [   22.044288] bbswitch: Succesfully loaded. Discrete card 0000:01:00.0 is on
> [   22.045583] bbswitch: disabling discrete graphics
> [   22.045690] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20160831/nsarguments-95)
> [   25.444627] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
> [   25.517205] wlan0: authenticate with f4:ec:38:ac:21:cc
> [   25.523150] wlan0: send auth to f4:ec:38:ac:21:cc (try 1/3)
> [   25.524897] wlan0: authenticated
> [   25.529116] wlan0: associate with f4:ec:38:ac:21:cc (try 1/3)
> [   25.533137] wlan0: RX AssocResp from f4:ec:38:ac:21:cc (capab=0x431 status=0 aid=1)
> [   25.535719] wlan0: associated
> [   25.535800] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready

`uname -a` output:
> Linux verter 4.9.0-kali3-amd64 #1 SMP Debian 4.9.18-1kali1 (2017-04-04) x86_64 GNU/Linux

Kali Linux 2017.1
Nvidia driver version 375.39

`Xorg -version` output:
> X.Org X Server 1.19.2
> Release Date: 2017-03-02
> X Protocol Version 11, Revision 0
> Build Operating System: Linux 3.16.0-4-amd64 x86_64 Debian
> Current Operating System: Linux verter 4.9.0-kali3-amd64 #1 SMP Debian 4.9.18-1kali1 (2017-04-04) x86_64
> Kernel command line: BOOT_IMAGE=/boot/vmlinuz-4.9.0-kali3-amd64 root=UUID=c52abde7-96bc-4f48-b2aa-f09919ab704d ro initrd=/install/gtk/initrd.gz pci=noaer acpi_osi=! "acpi_osi=Windows 2009" acpi_enforce_resources=lax
> Build Date: 03 March 2017  03:14:41PM
> xorg-server 2:1.19.2-1 (https://www.debian.org/support) 
> Current version of pixman: 0.34.0

ACPI-info script result:
[Micro-Star_International_Co.,_Ltd.-GL62_6QD.tar.gz](https://github.com/Bumblebee-Project/bbswitch/files/905159/Micro-Star_International_Co._Ltd.-GL62_6QD.tar.gz)

Vendor's site:
https://www.msi.com/Laptop/GL62-6QD.html#hero-overview

`cat /proc/acpi/dump_info` output:
> 0000:00:00.0 060000 \_SB_.PCI0.DSC0
> 0000:00:01.0 060400 \_SB_.PCI0.PEG0
> 0000:00:02.0 030000 \_SB_.PCI0.GFX0
> 0000:00:14.0 0c0330 \_SB_.PCI0.XHC_
> 0000:00:14.2 118000 
> 0000:00:16.0 078000 \_SB_.PCI0.HECI
> 0000:00:17.0 010601 \_SB_.PCI0.SAT0
> 0000:00:1c.0 060400 \_SB_.PCI0.RP01
> 0000:00:1c.3 060400 \_SB_.PCI0.RP04
> 0000:00:1f.0 060100 \_SB_.PCI0.LPCB
> 0000:00:1f.2 058000 \_SB_.PCI0.PPMC
> 0000:00:1f.3 040300 \_SB_.PCI0.HDAS
> 0000:00:1f.4 0c0500 \_SB_.PCI0.SBUS
> 0000:01:00.0 030200 \_SB_.PCI0.PEG0.PEGP
> 0000:02:00.0 028000 \_SB_.PCI0.RP01.PXSX
> 0000:03:00.0 020000 \_SB_.PCI0.RP04.PXSX

**Comment by Lekensteyn** [comment]
If you don't use Bumblebee, do you have problems? What if you use the nouveau driver instead of the proprietary driver? And note that in both cases you might need the `acpi_osi` workaround because of #140 and https://github.com/Bumblebee-Project/Bumblebee/issues/764

**Comment by darknyaz** [comment]
It seems that I have problem with fan in all cases. I don't remember what was when I've just installed OS, but to successfully start sddm I had to add `nouveau.modeset=0`. nouveau also report such error:
> nouveau 0000:01:00.0: Refused to change power state, currently in D3
I just didn't try nouveau + bumblebee combination, should I?
I also don't need to add `acpi_osi=! "acpi_osi=Windows 2009"` options when using nouveau.
